lib/logstash/inputs/jdbc.rb

```diff
@@ -232,4 +246,30 @@ def update_state_file
     end
   end
 
+  private
+
+  def enable_encoding
+    !@default_encoding.nil? || (@default_encoding.nil? && !@columns_to_encode.empty?)
+  end
+
+  # make sure the encoding is uniform over fields
+  def convert(key, value)
+    return value unless should_convert(key, value)
+    converter = find_converter_for(key, value)
+    converter.convert(value)
+  end
+
+  def should_convert(key, value)
+    column_keys = @columns_to_encode.keys
+    value.is_a?(String) && (!@default_encoding.nil? || column_keys.include?(key))
+  end
+
+  def find_converter_for(key, value)
+    encoding = @columns_to_encode[key] || @default_encoding
+    unless @converters.keys.include?(encoding)
+      @converters[encoding] = LogStash::Util::Charset.new(encoding)
+    end
+    @converters[encoding]
+  end
```

Comment:
Here's a different strategy you could look at: instead of having this `should_convert` / `find_converter_for` dance for all columns and row, just create a conversion strategy for **all** columns with one possible strategy being a "no conversion" or "identity". This conversion strategy for all columns could be initialized at the very first result row, and then simply systematically applied to all other rows and it would be a simple loop like this:

```
transcoded_row = row.inject({}) do |result, (k, v)|
  result[k] = @columns_converters[k].call(v)
  result
end
```

or something like that?